### DISCUSSION TITLE
How to set up group associations

### DISCUSSION THREAD

**Discussion by MikeyMan83** [zwave-js/zwave-js-ui#3035] [General]
Hi There,

In order to be less reliant on domoticz being up and running, i used to use group associations in OpenZwave before.
These groups seem to have been 'copied' to the ZwaveJS install somehow, but they don't seem to work.

A scenario of what i'm trying to do:
I have a battery powered 8 button zwave switch. I used to have 1 button controlling the dining room. With group association i was able to dim the light with keeping the on and off buttons pressed. This is a hell of a job if you want to do that with scenes.
On the receiving end i have Fibaro FGD212 Dimmers. 

How do i setup this grouping again / correctly? Do i add it to the switch or to the dimmer? 

Looking at the settings for the fibaro, i cannot group to a specific button or group:
![image](https://user-images.githubusercontent.com/6270724/229797049-5c2fb450-9158-460d-b853-92fd5bb6e7e8.png)

And looking at the switch, i cannot group to the specific endpoint:
![image](https://user-images.githubusercontent.com/6270724/229797257-ea67d935-b257-4987-9e93-8f59108e8f84.png)

**Comment by robertsLando** [comment]
You have to go to the switch node and add the light to the correct group that controls button1 state. In general keep in mind that associations works in this way:

I add node A to node B group X when I want node B to send something to A

**Comment by MikeyMan83** [comment]
Managed to get it working. Since it's a battery operated switch; it has to be set to 'learning mode' in order to do any grouping.

![image](https://user-images.githubusercontent.com/6270724/230038738-74590c8b-702e-4a48-a343-79674889ea70.png)

No clue what the function of that endpoint is, and descriptions could be a little better, but that's probably coming directly from the device itself.

**Reply by robertsLando** [reply]
Maybe @AlCalzone can help explaining you that

**Reply by AlCalzone** [reply]
Devices can have multiple endpoints, e.g. two-channel switches. What each endpoint represents must be explained in the manual. The root device (endpoint 0, or None) may control a single endpoint or multiple ones at the same time. Again, explained in the manual.

We have some devices where we added labels for the endpoints, but sadly those are still scarce.

The difference between target endpoint 0 and none is technical, and depends on whether the controlling device can send commands to endpoints or not. They often mean the same but some devices may behave differently.
